src/lua/Cloud.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "Cloud.hpp"
+#include "Util.hxx"
+#include "Interface.hpp"
+#include "Dialogs/Cloud/WeGlideUploadDialog.hpp"
+
+extern "C" {
+#include <lauxlib.h>
+}
+
+static int
+l_cloud_index(lua_State *L) {
+  return 1;
+}
+
+
+static int
+l_igcfile_upload(lua_State *L) {
+  if (lua_gettop(L) == 0 || lua_gettop(L) > 5)
+    return luaL_error(L, "Invalid parameters");
+  if (StringIsEqualIgnoreCase(luaL_checkstring(L, 1), "weglide")) {
+    StaticString<0x200> str;
+    str.SetASCII(luaL_checkstring(L, 2));
+
+    // Pilot ID:
+    WeGlide::Pilot pilot = {0, {0, 0, 0}};
+    pilot.id = lua_gettop(L) > 2 ? luaL_checknumber(L, 3) : 0;
+    // Pilot Date of Birth:
+    if (lua_gettop(L) > 3) {
+      unsigned day = 0, month = 0, year = 0;
+      // Date format should be ISO8601 (2000-01-01)
+      if (sscanf(luaL_checkstring(L, 4), "%04u-%02u-%02u", &year, &month,
+                 &day) == 3) { // sscanf 3 parameter
+        pilot.birthdate = {year, month, day};
+      }
+    }
+    // Glider ID:
+    uint_least32_t glider_id = lua_gettop(L) > 4 ? luaL_checknumber(L, 5) : 0;
+
+    if (WeGlide::UploadFlightDialog(Path(str), pilot, glider_id)) {
```

Comment:
This Lua function is undocumented, but its name does not seem to imply that it opens a modal dialog, and doing so doesn't sound like a good design.

Reply:
Oops, through this hint I noticed that WeGlide Update Flight Dialog is not a (real) dialog - here the IGC file is sent without a dialog window and a message box about success or failure is only displayed at the end ;-)
Therefore, in my opinion, this function should migrate from the dialog folder to the cloud folder and be called differently, e.g. WeGlide::UploadIGCFile (...).
@MaxKellermann : What do you think about that? BTW. I would postbone this lua point afterwards because at 1st I want to finishing the Upload procedure itself - lua documention makes additional work too...

Reply:
If it's not a dialog, it should not called "dialog" and the code should not be in the "Dialogs" folder. Isn't that obvious?

Reply:
Of course, but I just wanted to hear your thoughts on it one more time ...

Reply:
Documented in lua.rst